php/class-fieldmanager-field.php

```diff
@@ -1399,4 +1420,54 @@ public function escape( $field, $default = 'esc_html' ) {
 			return call_user_func( $default, $this->$field );
 		}
 	}
+
+	/**
+	 * Gets the JSON Schema for the field.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @see http://json-schema.org/draft-04/schema#
+	 *
+	 * @return array The JSON schema, represented as a PHP array.
+	 */
+	public function get_schema() {
+		$this->create_schema();
+		return $this->schema;
+	}
+
+	/**
+	 * Creates the JSON Schema for the field.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @see http://json-schema.org/draft-04/schema#
+	 */
+	protected function create_schema() {
+		/*
+		 * Set properties that map directly to JSON Schema properties or that are
+		 * required by the REST API.
+		 */
+		$this->schema['type'] = 'object';
+		$this->schema['description'] = $this->description;
+		$this->schema['context'] = array( 'view', 'edit' );
+	}
+
+	/**
+	 * Determine if this field can be shown in the REST API.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return bool True or false if the field can be shown in the REST API.
+	 */
+	public function can_show_in_rest() {
+		/**
+		 * Determine whether this field should be shown within the REST API.
+		 *
+		 * @since 1.3.0
+		 *
+		 * @param bool               $this->show_in_rest The current show in rest boolean.
+		 * @param Fieldmanager_Field                     The current field object.
+		 */
+		return apply_filters( 'fm_can_show_in_rest', $this->show_in_rest, $this );
```

Comment:
The more I think about it, the more I wonder whether the this filter is an invitation to accidentally expose sensitive data. All a developer would need to do is forget that they passed `__return_true` to the filter after a few months and then add a meta box or an option that managed the sensitive data. Thoughts?

Reply:
My intention was to give developers the ability to expose all fields to the REST API via a single line, and this filter allows them to do that.  I think exposing all fields and then blacklisting the ones that contain sensitive information is a good start.  Overall, it is still up to the developer to make sure they content exposed in the REST API is safe and secure, so I am ok with the way it is.  Though, as always, I would love to hear your thoughts :)